### PR TITLE
PS-581 Move from Auth to Keycloak, upgrade to Symfony 6, API Platform 3

### PR DESCRIPTION
**Pull request by aynsix**
- bump symfony version to symfony 6 for all app in phrasea:

 auth
databox
expose
notify
uploader
and all bundles in lib directory

### CODE REVIEW

File: lib/php/remote-auth-bundle/Security/Firewall/RemoteAuthListener.php [hunk 1]
```diff
@@ -1,56 +1,56 @@
...
+//namespace Alchemy\RemoteAuthBundle\Security\Firewall;
+//
+//use Alchemy\RemoteAuthBundle\Security\RequestHelper;
+//use Alchemy\RemoteAuthBundle\Security\Token\RemoteAuthToken;
+//use Symfony\Component\HttpKernel\Event\RequestEvent;
+//use Symfony\Component\Security\Core\Authentication\AuthenticationManagerInterface;
+//use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+//use Symfony\Component\Security\Core\Exception\AuthenticationException;
+//
+//class RemoteAuthListener
```
Comment (4rthem) [inline lib/php/remote-auth-bundle/Security/Firewall/RemoteAuthListener.php:14]: TODO

File: lib/php/remote-auth-bundle/Security/LoginFormAuthenticator.php [hunk 2]
```diff
@@ -44,109 +41,94 @@ class LoginFormAuthenticator extends AbstractFormLoginAuthenticator
...
     ) {
         $this->entityManager = $entityManager;
         $this->router = $router;
-        $this->csrfTokenManager = $csrfTokenManager;
-        $this->passwordEncoder = $passwordEncoder;
         $this->client = $client;
         $this->clientId = $clientId;
         $this->clientSecret = $clientSecret;
-        $this->session = $session;
+        $this->session = $requestStack->getSession();
```
Comment (4rthem) [inline lib/php/remote-auth-bundle/Security/LoginFormAuthenticator.php:57]: same

File: lib/php/remote-auth-bundle/Security/Provider/RemoteAuthProvider.php [hunk 3]
```diff
@@ -46,7 +45,7 @@ public function authenticate(TokenInterface $token)
 
         $authenticatedToken = new RemoteAuthToken($token->getAccessToken(), $roles);
         $authenticatedToken->setScopes($tokenInfo['scopes']);
-        $authenticatedToken->setAuthenticated(true);
+//        $authenticatedToken->setAuthenticated(true);
```
Comment (4rthem) [inline lib/php/remote-auth-bundle/Security/Provider/RemoteAuthProvider.php:48]: can be removed

File: lib/php/remote-auth-bundle/Security/RemoteAuthAuthenticator.php [hunk 4]
```diff
@@ -20,11 +21,11 @@ class RemoteAuthAuthenticator
 
     public function __construct(
         TokenStorageInterface $tokenStorage,
-        SessionInterface $session,
+        RequestStack $requestStack,
         EventDispatcherInterface $eventDispatcher
     ) {
         $this->tokenStorage = $tokenStorage;
-        $this->session = $session;
+        $this->session = $requestStack->getSession();
```
Comment (4rthem) [inline lib/php/remote-auth-bundle/Security/RemoteAuthAuthenticator.php:28]: same

File: lib/php/report-bundle/.php-cs-fixer.cache
Comment (4rthem) [inline lib/php/report-bundle/.php-cs-fixer.cache:1]: gitignore

File: lib/php/report-bundle/.php-cs-fixer.cache [hunk 5]
```diff
@@ -0,0 +1 @@
..._blank_line_before_namespace":true,"single_trait_insert_per_statement":true,"ternary_operator_spaces":true,"visibility_required":true,"blank_line_after_namespace":true,"constant_case":true,"control_structure_braces":true,"control_structure_continuation_position":true,"elseif":true,"function_declaration":true,"indentation_type":true,"line_ending":true,"lowercase_keywords":true,"no_break_comment":true,"no_closing_tag":true,"no_multiple_statements_per_line":true,"no_space_around_double_colon":true,"no_spaces_after_function_name":true,"no_spaces_inside_parenthesis":true,"no_trailing_whitespace":true,"no_trailing_whitespace_in_comment":true,"single_blank_line_at_eof":true,"single_line_after_imports":true,"statement_indentation":true,"switch_case_semicolon_to_colon":true,"switch_case_space":true,"encoding":true,"full_opening_tag":true},"hashes":{"Test\/MockHandlerFactory.php":"43ec652ebb7db1f21d5607f194f01c4d","ReportUserService.php":"1651ba96e2a8881d531a817207fda919","AlchemyReportBundle.php":"5b8286416672ba2feb3d43f55ab80c7e","DependencyInjection\/Configuration.php":"3ef553b9b63f1f059f7e4a03d204fca6","DependencyInjection\/AlchemyReportExtension.php":"61b3b0dbd66b93a7fe97e27941ea403d"}}
```
Comment (4rthem) [inline lib/php/report-bundle/.php-cs-fixer.cache:1]: gitignore

File: lib/php/storage-bundle/.php-cs-fixer.cache [hunk 6]
```diff
@@ -0,0 +1 @@
..."no_spaces_after_function_name":true,"no_spaces_inside_parenthesis":true,"no_trailing_whitespace":true,"no_trailing_whitespace_in_comment":true,"single_blank_line_at_eof":true,"single_line_after_imports":true,"statement_indentation":true,"switch_case_semicolon_to_colon":true,"switch_case_space":true,"encoding":true,"full_opening_tag":true},"hashes":{"Cdn\/CloudFrontUrlGenerator.php":"0dc0aacf4dfc41efe4270e7559f2c989","AlchemyStorageBundle.php":"3e4a43f3fd8e4dc9d94b5b1f180899ef","Command\/PruneMultipartUploadsCommand.php":"4675717ca52e7e3e24f6e0f337884dc8","Storage\/UrlSigner.php":"4486d765d6952a6b635c36aa9982768e","Storage\/FileStorageManager.php":"0aa2b0850edd23b44cd7086804c2a166","Storage\/PathGenerator.php":"7db41ada7b669bc8a36444424690cda2","Upload\/UploadManager.php":"5bc91dcab67fc0f45376c47453da3e8c","Entity\/MultipartUpload.php":"d02980239962a9bf81e0dfc69665cd88","Doctrine\/MultipartUploadListener.php":"98e2220a7239d28df574fe02b72308eb","DependencyInjection\/Configuration.php":"7547e61c2919c4fc6579e066b3f679fd","DependencyInjection\/AlchemyStorageExtension.php":"742b034232153dcb5e03af6fd8c0b6f1","Controller\/MultipartUploadPartAction.php":"477f7692f19fa82709a12dc89642da1f"}}
```
Comment (4rthem) [inline lib/php/storage-bundle/.php-cs-fixer.cache:1]: gitignore

File: lib/php/test-bundle/.php-cs-fixer.cache [hunk 7]
```diff
@@ -0,0 +1 @@
...ent":true,"ternary_operator_spaces":true,"visibility_required":true,"blank_line_after_namespace":true,"constant_case":true,"control_structure_braces":true,"control_structure_continuation_position":true,"elseif":true,"function_declaration":true,"indentation_type":true,"line_ending":true,"lowercase_keywords":true,"no_break_comment":true,"no_closing_tag":true,"no_multiple_statements_per_line":true,"no_space_around_double_colon":true,"no_spaces_after_function_name":true,"no_spaces_inside_parenthesis":true,"no_trailing_whitespace":true,"no_trailing_whitespace_in_comment":true,"single_blank_line_at_eof":true,"single_line_after_imports":true,"statement_indentation":true,"switch_case_semicolon_to_colon":true,"switch_case_space":true,"encoding":true,"full_opening_tag":true},"hashes":{"AlchemyTestBundle.php":"b9e66f9978f4460addcb423c574525ca","Mock\/EventProducerMock.php":"cdc1f734bb79a873191fb19f76b44b24","Helper\/FixturesTrait.php":"af4c43cf622349af63e384fa1f2bd82c","Helper\/TestServicesTrait.php":"9580ddbfd421532e6c22ff9f268fd136","DependencyInjection\/AlchemyTestExtension.php":"210440794762dc45030e343f38ded0ce","DependencyInjection\/Configuration.php":"75ea18929a69c99b99df928bd4dddbdd"}}
```
Comment (4rthem) [inline lib/php/test-bundle/.php-cs-fixer.cache:1]: gitignore

File: lib/php/webhook-bundle/.php-cs-fixer.cache [hunk 8]
```diff
@@ -0,0 +1 @@
...tsChoiceField.php":"7f205841efde2a93a4e4a208f918afa7","Listener\/TerminateStackListener.php":"bd0db10b4a767895b9df15d8cbb0d5b1","Consumer\/WebhookHandler.php":"13e10216e6b063b1e7b94043bb6966f0","Consumer\/SerializeObjectHandler.php":"95747f5ed3462a8a495c729b97eca10e","Consumer\/WebhookTriggerHandler.php":"9e12c776483b6332e0f587c4bff8cbed","Normalizer\/NormalizerContextBuilderInterface.php":"b3ada8b53af257a5471a7ab85080b1e7","Normalizer\/NormalizerContextBuilder.php":"529d76a1ca2a0fdbd5b7c7e0b7752771","Webhook\/WebhookTrigger.php":"dca43424cf97ddb6e4d0ddede1e79c6a","Webhook\/ObjectNormalizer.php":"c161059562db7de3d362bc5bbdcc401d","Entity\/Webhook.php":"c37b90fc3f378d10dd3d69b4316dc827","Entity\/WebhookLog.php":"4ff572efe34a85bed06ede667817d0fb","Doctrine\/Listener\/EntityListener.php":"b2bc12b2ae952d2e03d586be8bc1d3b4","Doctrine\/EntitySerializer.php":"05d2d0d1c00f324072757d70e03a4d71","DependencyInjection\/Configuration.php":"2fad90d1890afb623c588e51ba743bd9","DependencyInjection\/AlchemyWebhookExtension.php":"067a49e15c2f89e99f3e9e115e6d8ac2","Controller\/AdminControllerTrait.php":"94bba0331f9cfa036621317398753859","Config\/EntityRegistry.php":"0056e2cb31cfbb8b51b4bcdec92e99a3"}}
```
Comment (4rthem) [inline lib/php/webhook-bundle/.php-cs-fixer.cache:1]: gitignore

File: lib/php/workflow-bundle/.php-cs-fixer.cache [hunk 9]
```diff
@@ -0,0 +1 @@
...if":true,"function_declaration":true,"indentation_type":true,"line_ending":true,"lowercase_keywords":true,"no_break_comment":true,"no_closing_tag":true,"no_multiple_statements_per_line":true,"no_space_around_double_colon":true,"no_spaces_after_function_name":true,"no_spaces_inside_parenthesis":true,"no_trailing_whitespace":true,"no_trailing_whitespace_in_comment":true,"single_blank_line_at_eof":true,"single_line_after_imports":true,"statement_indentation":true,"switch_case_semicolon_to_colon":true,"switch_case_space":true,"encoding":true,"full_opening_tag":true},"hashes":{"Cache\/WorkflowCacheWarmer.php":"f912bacf2b3bb75edb17223a66c75899","Executor\/Adapter\/Action\/ServiceActionRegistry.php":"5744736859024b7b2e40b1b9796f20cc","AlchemyWorkflowBundle.php":"55b41ca03d62e3ce8db72e47a5e04df2","Doctrine\/EntityLoadListener.php":"dcecc5281570c9f92b8815723021c158","DependencyInjection\/AlchemyWorkflowExtension.php":"6f27e326b1eac8f1a03c095211bfa6f0","DependencyInjection\/Compiler\/WorkflowActionCompilerPass.php":"ee0bcd216125be90ae3e6c135b0cbbc3","DependencyInjection\/Configuration.php":"594f77b80d943a787566dbd75a7c65eb","Loader\/KernelCacheLoader.php":"1e5d6e78b5044744b2290c1ba145330a"}}
```
Comment (4rthem) [inline lib/php/workflow-bundle/.php-cs-fixer.cache:1]: gitignore

**Review by 4rthem — CHANGES_REQUESTED** [review]

File: auth/api/config/packages/security.yaml [hunk 10]
```diff
@@ -26,54 +25,49 @@ security:
         admin:
             pattern:    ^/admin
             stateless:  false
-            anonymous:  ~
             user_checker: App\Security\UserChecker
             logout:
                 path: alchemy_admin_logout
                 target: easyadmin
-                success_handler: alchemy_admin.logout.success_handler
```
[the remark this answers is not in the document]
Comment (aynsix) [reply auth/api/config/packages/security.yaml:34]: NO , in sf6 there is no more a success_handler, it's remove as a listener

**Review by 4rthem — APPROVED** [review]

**Pull request merged**